postgresql-ballerina/client.bal

```diff
@@ -0,0 +1,253 @@
+#                    Value of zero means that it is disabled.
+# + socketTimeout - The timeout value used for socket read operations.
+#                   If reading from the server takes longer than this value, the connection is close
+#                   Value of zero means that it is disabled.
+# + loginTimeout - Specify how long to wait for establishment of a database connection.
+#                  Value of zero means that it is infinite.
+# + rowFetchSize - Determine the number of rows fetched in ResultSet by one fetch with trip to the database.
+# + cachedMetadataFieldsCount - Specifies the maximum number of fields to be cached per connection.
+#                           A value of 0 disables the cache.
+# + cachedMetadataFieldSize - Specifies the maximum size (in megabytes) of fields to be cached per connection. 
+#                            A value of 0 disables the cache.
+# + preparedStatementThreshold - Determine the number of PreparedStatement executions required before switching over to use 
+#                            server side prepared statements.
+# + preparedStatementCacheQueries - Determine the number of queries that are cached in each connection.
+# + preparedStatementCacheSize - Determine the maximum size (in mebibytes) of the prepared queries.
+# + cancelSignalTimeout - Cancel command is sent out of band over its own connection, so cancel 
+#                         message can itself get stuck. So the timeout seconds for that.
+#                         Default value is 10 seconds
+# + keepAliveTcpProbe - Enable or disable TCP keep-alive probe.
+# + binaryTransfer - Use binary format for sending and receiving data if possible
+
+public type Options record {|
+  SecureSocket ssl = {};
+  decimal connectTimeout = 0;
+  decimal socketTimeout = 0;
+  decimal loginTimeout = 0;
+  int rowFetchSize?;
+  int cachedMetadataFieldsCount?;
+  int cachedMetadataFieldSize?;
+  int preparedStatementThreshold?;
+  int preparedStatementCacheQueries?;
+  int preparedStatementCacheSize?;
+  decimal cancelSignalTimeout = 10;
+  boolean keepAliveTcpProbe?;
+  boolean binaryTransfer?;
+|};
+
+# Possible values for SSL mode.
+# 
+public enum SSLMode {
+   PREFER,
+   REQUIRE,
+   DISABLE,
+   ALLOW,
+   VERIFY_CA = "VERIFY-CA",
+   VERIFY_FULL = "VERIFY-FULL"
+}
+
+# SSL Configuration to be used when connecting to Postgresql server.
+#
+# + mode - `SSLMode` to be used during the connection
+# + key - Keystore configuration of the client certificates
+# + rootcert - File name of the SSL root certificate. Defaults to defaultdir/root.crt.
+#             where defaultdir is ${user.home}/.postgresql/ in unix systems and %appdata%/postgresql/ on windows.
+ 
+public type SecureSocket record {|
+    SSLMode mode = PREFER;
+    string rootcert?;
+    crypto:KeyStore | CertKey key?;
+|};
+
+# Represents combination of certificate, private key and private key password if encrypted.
```

Comment:
```suggestion
# Represents the combination of the certificate, private key, and private key password if encrypted.
```

Reply:
I updated. commit id is fb00e99b1e6ee7ff1be5ede4529d1571294c214b